includes/definitions/discovery/eaton-mgeups.yaml

```diff
@@ -173,4 +179,17 @@ modules:
                         - { descr: 'Tolerance Volt. Out', graph: 0, value: 2, generic: 2 }
                         - { descr: 'Tolerance Freq. Out', graph: 0, value: 3, generic: 2 }
                         - { descr: 'No Voltage', graph: 0, value: 4, generic: 2 }
+                -
+                    oid: xupsAlarms
+                    value: xupsAlarms
+                    num_oid: '.1.3.6.1.4.1.534.1.7.1.{{ $index }}'
+                    descr: 'XUPS Alarm'
+                    index: 'xupsAlarms.{{ $index }}'
+                    state_name: xupsAlarms
+                    states:
+                        - { descr: 'No Alarm', graph: 0, value: 0, generic: 0 }
+                        - { descr: 'Alarm raised', graph: 0, value: 1, generic: 2 }
+                        - { descr: 'Alarms raised', graph: 0, value: 2, generic: 2 }
+                        - { descr: 'Alarms raised', graph: 0, value: 3, generic: 2 }
+                        - { descr: 'Alarms raised', graph: 0, value: 4, generic: 2 }
```

Comment:
why not this way?
```
         count: 
                -
                    oid: xupsAlarms
                    value: xupsAlarms
                    num_oid: '.1.3.6.1.4.1.534.1.7.1.{{ $index }}'
                    descr: 'XUPS Alarm'
                    index: 'xupsAlarms.{{ $index }}'
                    high_limit: 1
```
I personally would think about warn_limit 1 and high_limit 2 because if you have an alarm you should get a yellow warning, in the most of the times. If this grows up you have bigger problems so you should get a red state but this depends on the hardware :-). A better way would be to walk thought the underlying table in the MIB and print the details.

Reply:
@mpikzink Thank you for the feedback. I believe the number of alarms from your UPS is a dubious measure for the urgency implied.  :-)    

And yes, parsing the MIB and printing more details would be more informative. But a)I do not entirely trust the reported details, seeing how these alarms are reported in the web ui of the device. And b) I did not trust my understanding of the mechanics involved.  The mib speaks of alarmID and alarmEventID and one is likely the unique alarm event and the other the type of alarm. And all in all, I just want LibreNMS to flag that there is *something* to look into.

Grouping the xups alarms with the other alarm-related sensors made sense to me. Hence the some what silly 'states' stanza.

Reply:
Hi @dagbdagb 
The issue I see here is if you have, let say, more than 5 alarms ... State conversion will fail. So I agree with @mpikzink , you need to use count, set a high limit at 1 (so anything except 0 will trigger an alert) and you'll end up with exactly the same information, without the risk of failure with the states.